bin/delete-shards-without-contracts.js

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const program = require('commander');
+const Storage = require('storj-service-storage-models');
+const mongoose = require('mongoose');
+const Config = require('../lib/config');
+
+// Example:
+// node bin/delete-shards-without-contracts.js \
+// -u mongodb://admin:password@localhost:27017/__inxt-network
+
+
+program.version('0.0.1')
+  .option('-c, --config <path_to_config_file>', 'path to the config file')
+  .option('-u, --mongourl <mongo connection string>', 'mongo url connection string')
+  .parse(process.argv);
+
+
+function deleteShardsWithoutContracts() {
+  if (!process.env.NODE_ENV) {
+    throw new Error('NODE_ENV is not set');
+  }
+  try {
+    const config = new Config(process.env.NODE_ENV, program.config);
+    const storage = new Storage(
+      program.mongourl || config.storage.mongoUrl,
+      config.storage.mongoOpts
+    );
+
+    const Shard = storage.models.Shard;
+
+    const cursor = Shard.find()
+      .sort({
+        '_id': 1
+      })
+      .cursor();
+
+    let chunkOfShards = [];
+    const chunkSize = 5;
+
+    cursor.on('data', (shard) => {
+      chunkOfShards.push(shard);
+      if (chunkOfShards.length === chunkSize) {
+        cursor.pause();
+      }
+    });
+
+    let idShardBeingChecked;
+
+    let deleteCount = 0;
+    let checkedShards = 0;
+
+    const checkAndDeleteShard = (shard, cb) => {
+      checkedShards += 1;
```

Comment:
Is a great idea to keep the count of checked and deleted count but try to do it outside this reusable method, therefore you will avoid problems like: "I have to call this method but I do not want to increase these counters" (in a retry for example)

Reply:
good idea, fixed